src/main/java/io/tiledb/java/api/Query.java

```diff
@@ -305,6 +305,68 @@ public synchronized Query setBuffer(String attr, NativeArray buffer) throws Tile
     return this;
   }
 
+  /**
+   * Sets a buffer for a fixed-sized attribute.
+   *
+   * @param attr The attribute name.
+   * @param buffer NativeBuffer to be used for the attribute values.
+   * @param bufferElements The actual number of buffer elements
+   * @exception TileDBError A TileDB exception
+   */
+  public synchronized Query setBuffer(String attr, NativeArray buffer, int bufferElements)
+      throws TileDBError {
+    if (bufferElements <= 0) {
+      throw new TileDBError("Number of buffer elements must be >= 1");
+    }
+    if (bufferElements > buffer.getSize()) {
+      throw new TileDBError(
+          "Number of elements requested exceeds the number of elements in allocated buffer: "
+              + bufferElements
+              + " > "
+              + buffer.getSize());
+    }
+
+    try (ArraySchema schema = array.getSchema()) {
+      if (attr.equals(tiledb.tiledb_coords())) {
+        try (Domain domain = schema.getDomain()) {
+          Types.typeCheck(domain.getType(), buffer.getNativeType());
+        }
+      } else {
+        try (Attribute attribute = schema.getAttribute(attr)) {
+          Types.typeCheck(attribute.getType(), buffer.getNativeType());
+        }
+      }
+    }
+
+    uint64_tArray offsets_array_size = new uint64_tArray(1);
+    uint64_tArray values_array_size = new uint64_tArray(1);
+
+    offsets_array_size.setitem(0, BigInteger.valueOf(0l));
+    values_array_size.setitem(0, BigInteger.valueOf(buffer.getNBytes()));
```

Comment:
You should set `bufferElements * buffer.getNativeTypeSize()` here so you only have to set it once instead of setting the size and updating it later.